newIDE/app/src/InstancesEditor/InstancesAdder.js

```diff
@@ -1,10 +1,13 @@
 // @flow
 import { roundPosition } from '../Utils/GridHelpers';
+import ObjectsAdditionalService from '../SceneEditor/ObjectsAdditionalService';
+import { type InfoBarDetails } from '../SceneEditor/ObjectsAdditionalService';
 const gd: libGDevelop = global.gd;
 
 type Props = {|
   instances: gdInitialInstancesContainer,
   options: Object,
+  onAdditionalServiceComplete: (infoBarDetails: InfoBarDetails) => void,
```

Comment:
I'm not a big fan of passing "onAdditionalServiceComplete":
- the name is not super helpful by itself ("additional service" is very generic),
- it adds the complexity of understanding what is "InfoBarDetails" to the reader of InstancesAdder. 
- More generally, this is "leaking" abstraction: as InstancesAdder, I should not even know that there is something called "SceneEditor" and I should not even worry about "ObjectsAdditionalService".

In other words, `SceneEditor` **uses** `InstancesEditor`, but not the other way around. 
If we introduce some imports/types/classes from `SceneEditor`, this is a sign that we're starting to make `InstancesEditor` dependend on `SceneEditor` - i.e: we're coupling them.

So how to solve this then? Give to your functions/classes/components only what they need and only what you need to do their job. Here, you want to react when an instance is added? 
Sure thing, just add a prop called "onInstanceAdded", which takes a gdInitialInstance as parameter. Then call it in InstancesAdder.
Then run the additional work *in the parent* (i.e: in SceneEditor), or more generally in the component that is responsible for reacting to this event.

Reply:
You can think of InstancesEditor as saying to its parent: "hey what's the deal with your info bar and your complex stuff that you imported from SceneEditor? Leave me alone, if you want to know when something happened than just tell me (with a prop/callback) and I'll tell you, but don't bother me doing your work, I'm already doing enough" 😄